packages/application-components/src/utils/default-form-buttons.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { useIntl, MessageDescriptor, IntlShape } from 'react-intl';
+import {
+  PrimaryButton,
+  SecondaryButton,
+  IconButton,
+  BinLinearIcon,
+} from '@commercetools-frontend/ui-kit';
+import buttonMessages from './button-messages';
+import filterDataAttributes from './filter-data-attributes';
+
+type Label = string | MessageDescriptor;
+type Props = {
+  label: Label;
+  onClick: (event: React.SyntheticEvent) => void;
+  isDisabled: boolean;
+  dataAttributes: { [key: string]: string };
+  children?: never;
+};
+
+const primaryDefaultProps: Pick<
+  Props,
+  'label' | 'isDisabled' | 'dataAttributes'
+> = {
+  label: buttonMessages.confirm,
+  isDisabled: false,
+  dataAttributes: {},
+};
+
+const getFormattedLabel = (label: Label, intl: IntlShape) =>
```

Comment:
Just a thought I had looking at this. Cause it could be:

```js
const useFormattedLabel = (label: Label) => {
   const intl = useIntl();

   return typeof label === 'string' ? label : intl.formatMessage(label);
}
```

then in code

```js
const formattedSecondaryButtonLabel = useFormattedLabel(props.label)
```